File: src/main/java/pl/dolien/shop/user/dto/UserRequestDTO.java
package pl.dolien.shop.user.dto;

import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserRequestDTO {
    private Integer id;
    private String firstname;
    private String lastname;
    private String email;
    private String password;
    private boolean accountLocked;
    private LocalDate dateOfBirth;
}

File: shop-ui/src/app/user/models/user-request-dto/user-request-dto.ts
import { Role } from '../../../role/models/role/role';

export class UserRequestDTO {
  constructor(
    public id: number = 0,
    public firstname: string = '',
    public lastname: string = '',
    public email: string = '',
    public password: string = '',
    public dateOfBirth: Date = new Date(),
    public accountLocked: boolean = false,
    public roles: Role[] = []
  ) {}
}
